### DISCUSSION TITLE
Delete all audit logs because my sqlite.db is very big (10GB)

### DISCUSSION THREAD

**Discussion by Alexconquer** [gravitational/teleport#29638] [Q&A] [answered]
Hello,

I have little VM for Teleport and my hdd is full.
But the sqlite.db is big, i think is content lot of audit logs.

I want delete audit logs, this is possible ?
I search in documentation, i nothing find.

i add this, audit_retention_period, on configuration file but no effect

Can you help me ?

Thanks

**Comment by webvictim** [comment]
Audit logs are actually stored under `/var/lib/teleport/log` - are you sure your sqlite database is 10GB?

**Reply by Alexconquer** [reply]
Yes i totaly sure

`# du -cksh *`
11G     backend
4.0K    host_uuid
290M    log
84K     proc
4.0K    webproxy_cert.pem
4.0K    webproxy_key.pem
11G     total

There is one file in backend : sqlite.db
I dont understand why... what is saved in it?

**Comment by Alexconquer** [comment]
I found the error in log of Teleport ! :D 
But I have no idea how to solve this.

`2023-07-27T09:39:20+02:00 WARN [SQLITE]    "Unexpected error in inTransaction: \nERROR REPORT:\nOriginal Error: sqlite3.Error database disk image is malformed\nStack Trace:\n\tgithub.com/gravitational/teleport/lib/backend/lite/periodic.go:119 github.com/gravitational/teleport/lib/backend/lite.(*Backend).removeOldEvents.func1\n\tgithub.com/gravitational/teleport/lib/backend/lite/lite.go:992 github.com/gravitational/teleport/lib/backend/lite.(*Backend).inTransaction\n\tgithub.com/gravitational/teleport/lib/backend/lite/periodic.go:110 github.com/gravitational/teleport/lib/backend/lite.(*Backend).removeOldEvents\n\tgithub.com/gravitational/teleport/lib/backend/lite/periodic.go:55 github.com/gravitational/teleport/lib/backend/lite.(*Backend).runPeriodicOperations\n\truntime/asm_amd64.s:1598 runtime.goexit\nUser Message: database disk image is malformed, rolling back." lite/lite.go:980
2023-07-27T09:39:20+02:00 WARN [SQLITE]    Failed to run remove old events: database disk image is malformed lite/periodic.go:57
2023-07-27T09:39:21+02:00 WARN [SQLITE]    "Unexpected error in inTransaction: \nERROR REPORT:\nOriginal Error: sqlite3.Error database disk image is malformed\nStack Trace:\n\tgithub.com/gravitational/teleport/lib/backend/lite/periodic.go:119 github.com/gravitational/teleport/lib/backend/lite.(*Backend).removeOldEvents.func1\n\tgithub.com/gravitational/teleport/lib/backend/lite/lite.go:992 github.com/gravitational/teleport/lib/backend/lite.(*Backend).inTransaction\n\tgithub.com/gravitational/teleport/lib/backend/lite/periodic.go:110 github.com/gravitational/teleport/lib/backend/lite.(*Backend).removeOldEvents\n\tgithub.com/gravitational/teleport/lib/backend/lite/periodic.go:55 github.com/gravitational/teleport/lib/backend/lite.(*Backend).runPeriodicOperations\n\truntime/asm_amd64.s:1598 runtime.goexit\nUser Message: database disk image is malformed, rolling back." lite/lite.go:980
2023-07-27T09:39:21+02:00 WARN [SQLITE]    Failed to run remove old events: database disk image is malformed lite/periodic.go:57`

**Comment by Alexconquer** [comment, marked as answer]
I solve it !!!!

i use this : https://docs.locusmap.app/doku.php?id=manual:advanced:map_tools:corrupted_sqlite

in resume : 

`# sqlite3 sqlite.db  "PRAGMA integrity_check;"`
*** in database main ***
On tree page 31 cell 195: Rowid 13515474 out of order
On tree page 31 cell 194: Rowid 13515470 out of order
On tree page 31 cell 193: Rowid 13515465 out of order
row 922 missing from index events_created
row 923 missing from index events_created
row 924 missing from index events_created
row 925 missing from index events_created
row 926 missing from index events_created
row 927 missing from index events_created
row 928 missing from index events_created
row 929 missing from index events_created
row 930 missing from index events_created
row 931 missing from index events_created
row 932 missing from index events_created
row 933 missing from index events_created
row 934 missing from index events_created
row 935 missing from index events_created

Next i dump the db to new db : 

`sqlite3 sqlite.db ".dump" | sqlite3 new-sqlite.db`

Make backup of sqlite.db before ;)

Replace file sqlite.db by new-sqlite.db

And start Teleport, at this moment, teleport is freeze during db processing (2h for me)

Finally i stop teleport and start cmd : `sqlite3 sqlite.db 'VACUUM;'` to free up disk space.

Now my sqlite.db is 500ko XD

**Reply by webvictim** [reply]
Nice! sqlite databases are unfortunately prone to corruption if the server reboots while data is being written, for example. I'm glad you managed to fix it :)
